### PR TITLE
fix: Avoid duplicates in form submission file

### PR DESCRIPTION
**Pull request by theus77**
|Q              |A  |
|---------------|---|
|Bug fix?       |Y|
|New feature?   |N|	
|BC breaks?     |N|	
|Deprecations?  |N|	
|Fixed tickets  |N|

### CODE REVIEW

File: Submission/FormDataFile.php [hunk 1]
```diff
@@ -34,12 +36,23 @@ public function getFile(): UploadedFile
...
     }
+
+    private function getFilename(UploadedFile $uploadedFile, string $fieldName)
+    {
+        $filename = $uploadedFile->getClientOriginalName();
+        $extension = MimeTypes::getDefault()->getExtensions($uploadedFile->getClientMimeType())[0] ?? null;
+        if ($extension !== null && !RequestRuntime::endsWith($filename, $extension)) {
+            $filename .= \sprintf('.%s', $extension);
+        }
+        return sprintf('%s.%s', \uniqid(sprintf('%s.', $fieldName), false), $filename);
```
Comment (Davidmattei) [inline Submission/FormDataFile.php:56]: \sprintf

File: Submission/FormDataFile.php [hunk 2]
```diff
@@ -34,12 +36,23 @@ public function getFile(): UploadedFile
...
-            'filename' => $this->file->getClientOriginalName(),
+            'filename' => $fileName,
             'pathname' => $this->file->getPathname(),
             'mimeType' => $this->file->getMimeType(),
             'size' => $this->file->getSize(),
             'form_field' => $this->formElement->getName()
         ];
     }
+
+    private function getFilename(UploadedFile $uploadedFile, string $fieldName)
```
Comment (Davidmattei) [inline Submission/FormDataFile.php:49]: return type string

**Review by Davidmattei — CHANGES_REQUESTED** [review]

File: Submission/FormDataFile.php (hunk 2 above)
Comment (theus77) [reply Submission/FormDataFile.php]: Fixed

File: Submission/FormDataFile.php (hunk 1 above)
Comment (theus77) [reply Submission/FormDataFile.php]: Fixed

**Review by Davidmattei — APPROVED** [review]

**Pull request merged**